File: shared/src/commonMain/kotlin/pe/ecolecta/domain/model/Auditoria.kt
package pe.ecolecta.domain.model

data class Auditoria(
    val id: String,
    val entidad: String,
    val entidadId: String,
    val accion: AccionAuditoria,
    val valorAntes: String?,
    val valorDespues: String?,
    val motivo: String?,
    val usuarioId: String,
    val ocurridoEn: Long,
    val deviceId: String,
    val syncState: SyncState,
)

File: web/app/Domain/Auditoria/Auditoria.php
<?php

namespace App\Domain\Auditoria;

use DateTimeImmutable;

final class Auditoria
{
    public function __construct(
        public readonly ?int $id,
        public readonly string $entidad,
        public readonly int $entidadId,
        public readonly AccionAuditoria $accion,
        public readonly ?string $valorAntes,
        public readonly ?string $valorDespues,
        public readonly ?string $motivo,
        public readonly int $usuarioId,
        public readonly DateTimeImmutable $ocurridoEn,
    ) {}
}
